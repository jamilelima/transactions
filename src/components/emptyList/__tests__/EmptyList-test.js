import React from 'react';
import EmptyList from '../index';
import {create} from 'react-test-renderer';
import {shallow} from 'enzyme';

describe('Test EmptyList component', () => {
  const emptyList = create(<EmptyList />);
  test('Match with snapashot', () => {
    expect(emptyList.toJSON()).toMatchSnapshot();
  });
  test('Test empty list text', () => {
    const component = shallow(<EmptyList />);
    expect(
      component.text().includes('A lista está vazia. Adicione uma transação.'),
    ).toBe(true);
  });
  test("Test empty list text for a text that doesn't exist", () => {
    const component = shallow(<EmptyList />);
    expect(
      component.text().includes('A lista está vazia. Adicione uma transação.'),
    ).toBe(true);
  });
});
