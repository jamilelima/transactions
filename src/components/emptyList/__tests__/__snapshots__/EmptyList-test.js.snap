// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test EmptyList component Match with snapashot 1`] = `
<View
  style={
    Array [
      Object {
        "alignContent": "center",
        "alignItems": "center",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
>
  <RNSVGSvgView
    bbHeight="70%"
    bbWidth="70%"
    focusable={false}
    height="70%"
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        Object {
          "flex": 0,
          "height": "70%",
          "width": "70%",
        },
      ]
    }
    width="70%"
  >
    <RNSVGGroup>
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight="100%"
        bbWidth="100%"
        data-name="Layer 1"
        focusable={false}
        height="100%"
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            Object {
              "flex": 0,
              "height": "100%",
              "width": "100%",
            },
          ]
        }
        vbHeight={780.81}
        vbWidth={820.16}
        width="100%"
      >
        <RNSVGGroup>
          <RNSVGDefs>
            <RNSVGLinearGradient
              gradient={
                Array [
                  0,
                  1082163328,
                  0.54,
                  528515200,
                  1,
                  444629120,
                ]
              }
              gradientTransform={
                Array [
                  1,
                  0,
                  0,
                  1,
                  -3.62,
                  1.57,
                ]
              }
              gradientUnits={1}
              name="prefix__a"
              x1={539.63}
              x2={539.63}
              y1={734.6}
              y2={151.19}
            />
            <RNSVGLinearGradient
              gradient={Array []}
              gradientTransform={
                Array [
                  1,
                  0,
                  0,
                  1,
                  -63.92,
                  7.85,
                ]
              }
              gradientUnits={0}
              name="prefix__b"
              x1={540.17}
              x2={540.17}
              y1={180.2}
              y2={130.75}
            />
            <RNSVGLinearGradient
              gradient={Array []}
              gradientTransform={
                Array [
                  0.9777466365253419,
                  -0.20978921507880455,
                  0.20978921507880455,
                  0.9777466365253419,
                  -84.50985011037324,
                  124.59998019074499,
                ]
              }
              gradientUnits={0}
              name="prefix__c"
              x1={540.17}
              x2={540.17}
              y1={140.86}
              y2={82.43}
            />
            <RNSVGLinearGradient
              gradient={Array []}
              gradientTransform={null}
              gradientUnits={0}
              name="prefix__d"
              x1={476.4}
              x2={476.4}
              y1={710.53}
              y2={127.12}
            />
            <RNSVGLinearGradient
              gradient={Array []}
              gradientTransform={null}
              gradientUnits={0}
              name="prefix__e"
              x1={476.94}
              x2={476.94}
              y1={156.13}
              y2={106.68}
            />
            <RNSVGLinearGradient
              gradient={Array []}
              gradientTransform={null}
              gradientUnits={0}
              name="prefix__f"
              x1={666.86}
              x2={666.86}
              y1={176.39}
              y2={117.95}
            />
          </RNSVGDefs>
          <RNSVGPath
            d="M69.12 135.49l427.295-91.682L623.09 634.19l-427.295 91.682z"
            fill={4292927712}
            propList={
              Array [
                "fill",
              ]
            }
          />
          <RNSVGPath
            d="M324.89 152.76h422.25v583.41H324.89z"
            fill={
              Array [
                1,
                "prefix__a",
              ]
            }
            matrix={
              Array [
                0.9777466365253419,
                -0.20978921507880455,
                0.20978921507880455,
                0.9777466365253419,
                -271.22007878572344,
                62.719956808077995,
              ]
            }
            opacity={0.5}
            propList={
              Array [
                "fill",
              ]
            }
          />
          <RNSVGPath
            d="M84.639 146.993L486.98 60.665l119.69 557.824-402.344 86.328z"
            fill={4294638330}
            propList={
              Array [
                "fill",
              ]
            }
          />
          <RNSVGPath
            d="M374.18 138.6h204.14v49.45H374.18z"
            fill={
              Array [
                1,
                "prefix__b",
              ]
            }
            matrix={
              Array [
                0.9777466365253419,
                -0.20978921507880455,
                0.20978921507880455,
                0.9777466365253419,
                -213.5800667868331,
                43.92985326802773,
              ]
            }
            propList={
              Array [
                "fill",
              ]
            }
          />
          <RNSVGPath
            d="M460.93 91.9c-15.41 3.31-25.16 18.78-21.77 34.55s18.62 25.89 34 22.58 25.16-18.78 21.77-34.55-18.59-25.89-34-22.58zm9.67 45.1a16.86 16.86 0 1112.56-20 16.66 16.66 0 01-12.56 20z"
            fill={
              Array [
                1,
                "prefix__c",
              ]
            }
            matrix={
              Array [
                1,
                0,
                0,
                1,
                -189.92,
                -59.59,
              ]
            }
            propList={
              Array [
                "fill",
              ]
            }
          />
          <RNSVGPath
            d="M183.007 98.422L378.4 56.498l9.917 46.218-195.393 41.924z"
            fill={4283652854}
            propList={
              Array [
                "fill",
              ]
            }
          />
          <RNSVGPath
            d="M271.01 32.31a27.93 27.93 0 1033.17 21.45 27.93 27.93 0 00-33.17-21.45zm9.24 43.1a16.12 16.12 0 1112.38-19.14 16.12 16.12 0 01-12.38 19.14z"
            fill={4283652854}
            propList={
              Array [
                "fill",
              ]
            }
          />
          <RNSVGPath
            d="M257.89 116.91h437.02v603.82H257.89z"
            fill={4292927712}
            propList={
              Array [
                "fill",
              ]
            }
          />
          <RNSVGPath
            d="M265.28 127.12h422.25v583.41H265.28z"
            fill={
              Array [
                1,
                "prefix__d",
              ]
            }
            opacity={0.5}
            propList={
              Array [
                "fill",
              ]
            }
          />
          <RNSVGPath
            d="M270.65 131.42h411.5v570.52h-411.5z"
            fill={4294967295}
            propList={
              Array [
                "fill",
              ]
            }
          />
          <RNSVGPath
            d="M374.87 106.68h204.14v49.45H374.87z"
            fill={
              Array [
                1,
                "prefix__e",
              ]
            }
            propList={
              Array [
                "fill",
              ]
            }
          />
          <RNSVGPath
            d="M666.86 118c-15.76 0-28.54 13.08-28.54 29.22s12.78 29.22 28.54 29.22 28.54-13.08 28.54-29.22S682.62 118 666.86 118zm0 46.08a16.86 16.86 0 1116.46-16.86A16.66 16.66 0 01666.86 164z"
            fill={
              Array [
                1,
                "prefix__f",
              ]
            }
            matrix={
              Array [
                1,
                0,
                0,
                1,
                -189.92,
                -59.59,
              ]
            }
            propList={
              Array [
                "fill",
              ]
            }
          />
          <RNSVGPath
            d="M377.02 104.56h199.84v47.27H377.02z"
            fill={4283652854}
            propList={
              Array [
                "fill",
              ]
            }
          />
          <RNSVGPath
            d="M476.94 58.41a27.93 27.93 0 1027.93 27.93 27.93 27.93 0 00-27.93-27.93zm0 44.05a16.12 16.12 0 1116.14-16.16 16.12 16.12 0 01-16.14 16.11z"
            fill={4283652854}
            propList={
              Array [
                "fill",
              ]
            }
          />
          <RNSVGGroup
            fill={4282902193}
            opacity={0.5}
            propList={
              Array [
                "fill",
              ]
            }
          >
            <RNSVGPath
              d="M15.27 737.05h3.76v21.33h-3.76z"
            />
            <RNSVGPath
              d="M27.82 745.84v3.76H6.49v-3.76z"
            />
          </RNSVGGroup>
          <RNSVGGroup
            fill={4282902193}
            opacity={0.5}
            propList={
              Array [
                "fill",
              ]
            }
          >
            <RNSVGPath
              d="M451.49 0h3.76v21.33h-3.76z"
            />
            <RNSVGPath
              d="M464.04 8.78v3.76h-21.33V8.78z"
            />
          </RNSVGGroup>
          <RNSVGPath
            d="M771.08 772.56a4.61 4.61 0 01-2.57-5.57 2.22 2.22 0 00.1-.51 2.31 2.31 0 00-4.15-1.53 2.22 2.22 0 00-.26.45 4.61 4.61 0 01-5.57 2.57 2.22 2.22 0 00-.51-.1 2.31 2.31 0 00-1.53 4.15 2.22 2.22 0 00.45.26 4.61 4.61 0 012.57 5.57 2.22 2.22 0 00-.1.51 2.31 2.31 0 004.15 1.53 2.22 2.22 0 00.26-.45 4.61 4.61 0 015.57-2.57 2.22 2.22 0 00.51.1 2.31 2.31 0 001.53-4.15 2.22 2.22 0 00-.45-.26z"
            fill={4283271920}
            opacity={0.5}
            propList={
              Array [
                "fill",
              ]
            }
          />
          <RNSVGPath
            d="M136.67 567.5a4.61 4.61 0 01-2.57-5.57 2.22 2.22 0 00.1-.51 2.31 2.31 0 00-4.15-1.53 2.22 2.22 0 00-.26.45 4.61 4.61 0 01-5.57 2.57 2.22 2.22 0 00-.51-.1 2.31 2.31 0 00-1.53 4.15 2.22 2.22 0 00.45.26 4.61 4.61 0 012.57 5.57 2.22 2.22 0 00-.1.51 2.31 2.31 0 004.15 1.53 2.22 2.22 0 00.26-.45 4.61 4.61 0 015.57-2.57 2.22 2.22 0 00.51.1 2.31 2.31 0 001.53-4.15 2.22 2.22 0 00-.45-.26zM665.08 68.18a4.61 4.61 0 01-2.57-5.57 2.22 2.22 0 00.1-.51 2.31 2.31 0 00-4.15-1.53 2.22 2.22 0 00-.26.45 4.61 4.61 0 01-5.57 2.57 2.22 2.22 0 00-.51-.1 2.31 2.31 0 00-1.53 4.15 2.22 2.22 0 00.45.26 4.61 4.61 0 012.57 5.57 2.22 2.22 0 00-.1.51 2.31 2.31 0 004.15 1.53 2.22 2.22 0 00.26-.45 4.61 4.61 0 015.57-2.57 2.22 2.22 0 00.51.1 2.31 2.31 0 001.53-4.15 2.22 2.22 0 00-.45-.26z"
            fill={4294826037}
            opacity={0.5}
            propList={
              Array [
                "fill",
              ]
            }
          />
          <RNSVGCircle
            cx={812.64}
            cy={314.47}
            fill={4294270788}
            opacity={0.5}
            propList={
              Array [
                "fill",
              ]
            }
            r={7.53}
          />
          <RNSVGCircle
            cx={230.73}
            cy={746.65}
            fill={4294270788}
            opacity={0.5}
            propList={
              Array [
                "fill",
              ]
            }
            r={7.53}
          />
          <RNSVGCircle
            cx={735.31}
            cy={477.23}
            fill={4294270788}
            opacity={0.5}
            propList={
              Array [
                "fill",
              ]
            }
            r={7.53}
          />
          <RNSVGCircle
            cx={87.14}
            cy={96.35}
            fill={4283271920}
            opacity={0.5}
            propList={
              Array [
                "fill",
              ]
            }
            r={7.53}
          />
          <RNSVGCircle
            cx={7.53}
            cy={301.76}
            fill={4282902193}
            opacity={0.5}
            propList={
              Array [
                "fill",
              ]
            }
            r={7.53}
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </RNSVGGroup>
  </RNSVGSvgView>
  <Text
    style={
      Array [
        Object {
          "color": "#90949c",
          "fontFamily": "Nunito-Regular",
          "fontSize": 12,
          "textAlign": "center",
        },
      ]
    }
  >
    A lista está vazia. Adicione uma transação.
  </Text>
</View>
`;
